GTScript-Parallel-model.md

```diff
@@ -103,3 +103,107 @@ for k in range(start, end):
         u[i, j, k] = 1
     parfor [i_start:i_end, j_start:j_end]:
         b[i, j, k] = u[i-2,j,k] + u[i+1,j,k] + u[i,j-1,k] + u[i,j-2,k]
+```
+
+## Conditionals
+
+### If-Statements
+
+- Conditions in `if` statements are required to be of scalar type, e.g. scalar variables (not fields) or scalar expressions (no fields involved).
+- Each statement inside the if and else branches is executed as a `parfor` loop (same as statements outside of branches).
+
+Rationale:
+- If the condition is on a field, it's extremely complex to catch all corner cases if the parfor-loop is applied on each statement. E.g. think about a case where the `if`-branch updates a field for some points which is read in the `else`-branch (requires buffering of the original input).
+- If the parfor would be applied outside of the `if`, the model would contain some unexpected limits:
+    1. Removing an `if(True)` will change behavior.
+    2. Changing an `if __INLINE(flag)` to a runtime `if(flag)` will change behavior.
+
+#### Example
+
+```python
+with computation(...):
+    with interval(start, end):
+        if(my_config_var):
+            a = 1
+            b = 2
+        else:
+            a = 2
+            b = 1
+```
+
+translates to 
+
+```python
+for k in range(start, end):
+    if(my_config_var):
+        parfor ij:
+            a = 1
+        parfor ij:
+            b = 2
+    else:
+        parfor ij:
+            a = 2
+        parfor ij:
+            b = 1
+```
+
+In a more complicated case with extended compute domain
+
+```python
+with computation(...):
+    with interval(start, end):
+        if(my_config_var):
+            tmp = in[-1,0,0]
+            tmp2 = tmp[-1,0,0]
+        else:
+            tmp2 = 1
+        out = tmp2[-1,0,0]
+```
+
+translates to 
+
+```python
+for k in range(start, end):
+    if(my_config_var):
+        parfor [i_start-2:i_end, j_start:j_end]:
+            tmp = in[-1,0,0]
+        parfor [i_start-1:i_end, j_start:j_end]:
+            tmp2 = tmp[-1,0,0]
+    else:
+        parfor [i_start-1:i_end, j_start:j_end]:
+            tmp2 = 1
+    parfor ij:
+        out = tmp2[-1,0,0]
+```
+
+The following case is illegal
```

Comment:
Again, we should allow fields.

But if we would not, I find the below really confusing, since it is not clear what is a field and what is not a field in our front end language.

Reply:
>But if we would not, I find the below really confusing, since it is not clear what is a field and what is not a field in our front end language.

This is a very different problem